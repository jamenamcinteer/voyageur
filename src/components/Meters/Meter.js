import React from "react";
import theme from "../../theme";
import styled from "styled-components";
import PropTypes from "prop-types";

const StyledMeter = styled.div`
  width: 100%;
  height: 10px;
  position: relative;
  background: #dce0e3;
  border-radius: 5px;
`;

const MeterProgress = styled.span`
  display: block;
  height: 100%;
  border-radius: 5px;
  position: relative;
  overflow: hidden;
`;

const Meter = props => {
  const handleOverviewProgressBar = () => {
    let actual = props.actual;
    let budgeted = props.budgeted;

    let width = (actual / budgeted) * 100;
    // let backgroundColor = props.theme.themeColor;
    let backgroundColor = theme.themeColor;

    if (width > 100) {
      width = 100;
      // backgroundColor = props.theme.themeColorRed;
      backgroundColor = theme.themeColorRed;
    }
    if (budgeted === 0) {
      width = 0;
    }
    width = width + "%";

    return { width, backgroundColor };
  };

  return (
    <StyledMeter>
      <MeterProgress style={handleOverviewProgressBar()} />
    </StyledMeter>
  );
};

Meter.propTypes = {
  actual: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  budgeted: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
};

export default Meter;
