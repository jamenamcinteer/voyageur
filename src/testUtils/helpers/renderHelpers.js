import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export function renderWithTheme(component) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}

export function renderWithReduxRouterAndTheme(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router history={history}>{ui}</Router>
        </ThemeProvider>
      </Provider>
    ),
    store,
    history
  };
}
