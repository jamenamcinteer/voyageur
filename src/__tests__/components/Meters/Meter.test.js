import React from "react";
import { cleanup } from "@testing-library/react";
import Meter from "../../../components/Meters/Meter";
import { renderWithTheme } from "../../../testUtils/helpers/renderHelpers";

afterEach(cleanup);

test("should render with theme and passed props as numbers", () => {
  const { asFragment } = renderWithTheme(<Meter actual={50} budgeted={100} />);
  expect(asFragment()).toMatchSnapshot();
});

test("should render with theme and passed props as strings", () => {
  const { asFragment } = renderWithTheme(<Meter actual="50" budgeted="100" />);
  expect(asFragment()).toMatchSnapshot();
});
