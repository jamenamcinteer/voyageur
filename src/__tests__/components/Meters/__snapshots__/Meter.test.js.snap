// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with theme and passed props as numbers 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 10px;
  position: relative;
  background: #dce0e3;
  border-radius: 5px;
}

.c1 {
  display: block;
  height: 100%;
  border-radius: 5px;
  position: relative;
  overflow: hidden;
}

<div
    class="c0"
  >
    <span
      class="c1"
      style="width: 50%; background-color: rgb(12, 124, 136);"
    />
  </div>
</DocumentFragment>
`;

exports[`should render with theme and passed props as strings 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  height: 10px;
  position: relative;
  background: #dce0e3;
  border-radius: 5px;
}

.c1 {
  display: block;
  height: 100%;
  border-radius: 5px;
  position: relative;
  overflow: hidden;
}

<div
    class="c0"
  >
    <span
      class="c1"
      style="width: 50%; background-color: rgb(12, 124, 136);"
    />
  </div>
</DocumentFragment>
`;
